### PR TITLE
Armcore defaults

### PR DESCRIPTION
**Pull request by jhendrixMSFT**
- [ ] The purpose of this PR is explained in this or a referenced issue.
- [ ] The PR does not update generated files.
   - These files are managed by the codegen framework at [Azure/autorest.go][].
- [ ] Tests are included and/or updated for code changes.
- [ ] Updates to [CHANGELOG.md][] are included.
- [ ] Apache v2 license headers are included in each file.
 
[Azure/autorest.go]: https://github.com/Azure/autorest.go
[CHANGELOG.md]: https://github.com/Azure/azure-sdk-for-go/blob/master/CHANGELOG.md

### CODE REVIEW

File: sdk/armcore/connection.go [hunk 1]
```diff
@@ -55,25 +50,39 @@ type Connection struct {
...
+		Retry:      options.Retry,
+		Telemetry:  options.Telemetry,
+	}
+	if options.DisableRPRegistration == true {
+		regRPOpts.MaxAttempts = -1
+	}
 	p := azcore.NewPipeline(options.HTTPClient,
 		azcore.NewTelemetryPolicy(&options.Telemetry),
-		NewRPRegistrationPolicy(endpoint, cred, &options.RegisterRPOptions),
+		NewRPRegistrationPolicy(endpoint, cred, &regRPOpts),
```
Comment (catalinaperalta) [inline sdk/armcore/connection.go:85]: If they choose to disable RP registration, shouldnt we just remove the policy from the pipeline and avoid any additional processing? (I know it's probably very minimal, but still why add a policy that wont be used.)

File: sdk/armcore/policy_register_rp.go [hunk 2]
```diff
@@ -26,62 +26,76 @@ const (
...
+	}
+	if r.PollingDelay == 0 {
+		r.PollingDelay = 15 * time.Second
+	} else if r.PollingDelay < 0 {
+		r.PollingDelay = 0
+	}
+	if r.PollingDuration == 0 {
+		r.PollingDuration = 5 * time.Minute
+	} else if r.PollingDuration < 0 {
+
```
Comment (catalinaperalta) [inline sdk/armcore/policy_register_rp.go:75]: I think something is missing here?

File: sdk/armcore/connection_test.go [hunk 3]
```diff
@@ -128,3 +131,40 @@ func TestScope(t *testing.T) {
...
+	srv.SetResponse(mock.WithStatusCode(http.StatusConflict), mock.WithBody([]byte(rpUnregisteredResp)))
+	con := NewConnection(srv.URL(), mockTokenCred{}, &ConnectionOptions{DisableRPRegistration: true})
+	if ep := con.Endpoint(); ep != srv.URL() {
+		t.Fatalf("unexpected endpoint %s", ep)
+	}
+	req, err := azcore.NewRequest(context.Background(), http.MethodGet, srv.URL())
+	if err != nil {
+		t.Fatalf("Unexpected error: %v", err)
+	}
+	resp, err := con.Pipeline().Do(req)
```
Comment (catalinaperalta) [inline sdk/armcore/connection_test.go:148]: Why is this called twice? (also on line 159)

File: sdk/armcore/policy_register_rp.go [hunk 4]
```diff
@@ -26,62 +26,76 @@ const (
...
+		options: *o,
 	}
-	p := azcore.NewPipeline(o.HTTPClient,
-		azcore.NewTelemetryPolicy(&o.Telemetry),
-		azcore.NewRetryPolicy(&o.Retry),
-		cred.AuthenticationPolicy(azcore.AuthenticationPolicyOptions{Options: azcore.TokenRequestOptions{Scopes: []string{endpointToScope(endpoint)}}}),
-		azcore.NewLogPolicy(nil))
-	return &rpRegistrationPolicy{endpoint: endpoint, pipeline: p, options: *o}
+	// init the copy
+	p.options.init()
```
Comment (catalinaperalta) [inline sdk/armcore/policy_register_rp.go:97]: Out of curiosity, why is the init() called after assigning the options values to the policies?

Comment (jhendrixMSFT) [reply sdk/armcore/policy_register_rp.go:97]: The `init()` is only applicable to the options specified to this policy, the other options get initialized in their respective policy constructors.

File: sdk/armcore/connection_test.go (hunk 3 above)
Comment (jhendrixMSFT) [reply sdk/armcore/connection_test.go]: Copy/paste error, fixed.

File: sdk/armcore/policy_register_rp.go (hunk 2 above)
Comment (jhendrixMSFT) [reply sdk/armcore/policy_register_rp.go]: Removed.

File: sdk/armcore/policy_register_rp.go (hunk 4 above)
Comment (catalinaperalta) [reply sdk/armcore/policy_register_rp.go:95]: Ah yes, gotcha. Thanks for explaining! 😅

**Review by catalinaperalta — APPROVED** [review]
LGTM

**Pull request merged**